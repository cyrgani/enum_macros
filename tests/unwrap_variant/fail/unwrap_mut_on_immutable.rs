use enum_macros::UnwrapVariant;

#[derive(UnwrapVariant)]
#[allow(dead_code)]
enum Test {
    #[unwrap(ref, mut)]
    A(String),
    #[unwrap(mut, ref)]
    B(usize),
}

fn main() {
    let x = Test::A(String::from("x"));
    x.unwrap_A_mut().push('r');
    assert_eq!(x.unwrap_A_ref(), "rx");
}
