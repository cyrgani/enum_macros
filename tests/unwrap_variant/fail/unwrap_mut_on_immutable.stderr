error[E0596]: cannot borrow `x` as mutable, as it is not declared as mutable
  --> tests/unwrap_variant/fail/unwrap_mut_on_immutable.rs:14:5
   |
14 |     x.unwrap_A_mut().push('r');
   |     ^ cannot borrow as mutable
   |
help: consider changing this to be mutable
   |
13 |     let mut x = Test::A(String::from("x"));
   |         +++
